### PR TITLE
[Woo POS] The UI of the cart items

### LINKED ISSUE #11785: [Woo POS] Implement UI of the cart items
Image

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #11785

### Description

The PR implements the UI of the cart items

### Testing information

* Validate the UI against Figma
* Try adding/removing items

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/1e5ab7ee-0db6-4a37-bb02-1e5b007bb6c1

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>WooPosFormatPrice</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt [hunk 1]
```diff
@@ -155,56 +199,68 @@ private fun CartToolbar(
...
+        val fallbackColor = Color.LightGray
+        AsyncImage(
+            model = ImageRequest.Builder(LocalContext.current)
+                .data(item.imageUrl)
+                .crossfade(true)
+                .build(),
+            fallback = ColorPainter(color = fallbackColor),
+            error = ColorPainter(color = fallbackColor),
+            placeholder = ColorPainter(color = fallbackColor),
+            contentDescription = "Product Image",
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt:225]: 💡 This should be extracted to res

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt [hunk 2]
```diff
@@ -122,7 +122,9 @@ class WooPosCartViewModel @Inject constructor(
                         if (_state.value.isOrderCreationInProgress) return@collect
 
                         val itemClicked = viewModelScope.async {
-                            repository.getProductById(event.productId)?.toCartListItem()!!
+                            val product = repository.getProductById(event.productId)!!
+                            val itemNumber = (_state.value.itemsInCart.maxOfOrNull { it.id.itemNumber } ?: 0) + 1
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt:126]: I'm curious, what's `itemNumber` for. Is it a unique key for LazyList items? Would that work the same if we used `product.remoteId` instead?

**Review by samiuelson — APPROVED** [review]
Looks good. I left some non-blocking questions. Please take a look.

Other than these, I think there are some details worth improving in the next iterations:
* Adding placeholder image/icon for products without an image url
* There is a weird effect of list items being visible behind bottom button's bottom right corner. See the video:

https://github.com/woocommerce/woocommerce-android/assets/4527432/d2b03eec-e4db-4a21-9c77-52c6ec6a95b7

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModel.kt:126]: > I'm curious, what's itemNumber for. Is it a unique key for LazyList items?

We have to know which exact item we should remove, not only the key, that's why so

>  Would that work the same if we used product.remoteId instead?

I think it won't, as we can have the same type of product multiple times in the cart

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartScreen.kt:225]: Yep! I have done this already in the following PR

**Pull request merged**